packages/components/src/components/Text/Text.tsx

```diff
@@ -0,0 +1,77 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { type HTMLAttributes, type DetailedHTMLProps, type ReactNode, forwardRef, type ForwardedRef } from 'react';
+
+import { type CommonAriaProps } from '#/helpers/aria';
+import { cn } from '#/utilities/styleUtilities';
+
+import styles from './styles/text.module.scss';
+
+type TextWeight = 'default' | 'strong' | 'x-strong';
+type TextSize = 'x-small' | 'small' | 'medium' | 'large';
+type TextColor = 'default' | 'weak' | 'x-weak' | 'disabled' | 'negative' | 'positive' | 'warning' | 'interactive';
+type BoxColor = 'neutral' | 'selected' | 'disabled' | 'positive' | 'negative' | 'warning';
+
+type TagType = 'a' | 'abbr' | 'address' | 'em' | 'label' | 'li' | 'span' | 'strong' | 'time' | 'p';
+
+export type TextProps<TTag extends TagType = 'span'> = CommonAriaProps &
+    DetailedHTMLProps<HTMLAttributes<TextElementType<TTag>>, TextElementType<TTag>> & {
```

Comment:
After second thought, not too much of a fan to expose _all_ the props available as it could make maintenance a pain, could we restrict it to only what we need at the moment (same with Heading)

Reply:
Will go back to the Header component in a separate PR.